packages/common/src/constructors.ts

```diff
@@ -33,18 +34,21 @@ export function createCustomCommon(
 }
 
 /**
- * Static method to load and set common from a geth genesis JSON
- * @param genesisJSON JSON of geth configuration
+ * Static method to load and set common from a geth genesis object
+ * @param gethGenesis GethGenesis object
  * @param  opts additional {@link GethConfigOpts} for configuring common
  * @returns Common
  */
 export function createCommonFromGethGenesis(
-  genesisJSON: any,
+  genesisJSON: GethGenesis,
   { chain, eips, genesisHash, hardfork, params, customCrypto }: GethConfigOpts,
 ): Common {
   const genesisParams = parseGethGenesis(genesisJSON, chain)
   const common = new Common({
-    chain: genesisParams,
+    chain: {
+      ...genesisParams,
+      name: genesisParams.name ?? 'Custom chain',
```

Comment:
I think this would also have worked:

```suggestion
      name: 'Custom chain',
      ...genesisParams
```

Reply:
I tried it out, and it actually leads to this error:
`'name' is specified more than once, so this usage will be overwritten.ts(2783)`